jsp/src/main/java/com/sun/ts/tests/jsp/common/client/AbstractUrlClient.java

```diff
@@ -64,6 +75,35 @@ protected void setTestProperties(WebTestCase testCase) {
     super.setTestProperties(testCase);
   }
 
+  private boolean isNullOrEmpty(String val) {
+    if (val == null || val.equals("")) {
+      return true;
+    }
+    return false;
+  }
+
+  public void setup() throws Exception {
+
+    String hostname = System.getProperty(SERVLETHOSTPROP).trim();
```

Comment:
same comment as here https://github.com/jakartaee/platform-tck/pull/1160#issuecomment-1567721896

host and port should come from Arquillian.

Reply:
Hi Olivier @olamy Thanks for the comment.
How can we set a custom value for ArquillianResource url  when executing the test from the tck runner ?

Reply:
Well maybe you can use per default the value coming from Arquillian except if there is any sys prop which can override the default value coming from Arquillian.
This sounds a good consensus ;) 
wdyt?

Reply:
That sounds ok to me. So I guess there is no other way to customise the value of ArquillianResource.

Reply:
> That sounds ok to me. So I guess there is no other way to customise the value of ArquillianResource.

It depends on your container.
With Jetty it's a matter of a value in a file (https://github.com/jetty-project/servlet-tck-run/blob/c5deb83b9e6e80daaef82cec2e332a1988579596/src/test/resources/arquillian.xml#L14)  here `0` means random port which is return to the tests with the injection of the URL.

For Glassfish, maybe you should have a look at the arquillian container implementation. But I guess it defaults to 8080 or 8090.

Reply:
> That sounds ok to me. So I guess there is no other way to customise the value of ArquillianResource.

It depends on your container.
With Jetty it's a matter of a value in a file (https://github.com/jetty-project/servlet-tck-run/blob/c5deb83b9e6e80daaef82cec2e332a1988579596/src/test/resources/arquillian.xml#L14)  here `0` means random port which is return to the tests with the injection of the URL.

For Glassfish, maybe you should have a look at the arquillian container implementation. But I guess it defaults to 8080 or 8090.

Reply:
> That sounds ok to me. So I guess there is no other way to customise the value of ArquillianResource.

It depends on your container.
With Jetty it's a matter of a value in a file (https://github.com/jetty-project/servlet-tck-run/blob/c5deb83b9e6e80daaef82cec2e332a1988579596/src/test/resources/arquillian.xml#L14)  here `0` means random port which is return to the tests with the injection of the URL.

For Glassfish, maybe you should have a look at the arquillian container implementation. But I guess it defaults to 8080 or 8090.

Reply:
Okey. For now I have made the changes to use the host/port from system property by default , use the values from Arquillian if properties is not set.

Reply:
These values always come from the Arquillian Connector for a specific Server/Runtime. Best not to use any system property. It's unneeded and extra code, that someone once must read and understand. I think it can even be wrong, as the connector should be in control, and a system property would override that. Therefor the connector and property would be out of sync.

Just as an example for the GlassFish connector those properties are natively set here: https://github.com/OmniFish-EE/arquillian-container-glassfish/blob/main/glassfish-managed/src/main/java/org/omnifaces/arquillian/container/glassfish/managed/GlassFishManagedContainerConfiguration.java#L89

Reply:
I wasn't able to get hostname(url.getHost()) and port(url.getPort()) when running with glassfish. Do you know if anything more required ; glassfish-runner/jsp-tck/pom.xml has the Glassfish runner , jsp/src/main/java/com/sun/ts/tests/jsp/common/client/AbstractUrlClient.java has the common code for obtaining the url as below.

```
@ArquillianResource
@OperateOnDeployment("_DEFAULT_")
public URL url;
```

Reply:
Maybe you can use this for inspiration?

https://github.com/javaee-samples/jakartaee-samples/blob/main/ee9-tck/servlet/security-basic/src/test/java/org/jakartaee8/servlet/servletrequest/asynccontext/SecurityBasicTest.java

It uses the same "url-client" of the old TCK and configures it with the Url obtained from Arquillian (via the GlassFish Container / Connector).

(p.s. maybe it's an idea to put the "url-client" from the old TCK in a separate project like was done here: https://github.com/javaee-samples/jakartaee-samples/tree/main/url-client ?)

Reply:
I have addressed this issue in the latest commit by using URL injected by Arquillian to determine host and port. Hopefully further improvements can be made in future changes.